src/com/lilithsthrone/game/combat/Combat.java

```diff
@@ -177,7 +179,40 @@ public void initialiseCombat(
 			}
 		}
 
-		combatContent = combatStringBuilder.toString();
+		Main.game.getPlayer().resetSelectedMoves();
+		Main.game.getPlayer().resetMoveCooldowns();
+		Main.game.getPlayer().setRemainingAP(Main.game.getPlayer().getMaxAP(), null, null);
+		for(NPC npc : allCombatants)
+		{
+			npc.resetSelectedMoves();
+			npc.resetDefaultMoves(); // Resetting in case the save file was too old and NPC has no moves selected for them.
+			npc.resetMoveCooldowns();
+			npc.setRemainingAP(npc.getMaxAP(), null, null);
+			// Sets up NPC ally/enemy lists that include player
+			List<GameCharacter> npcAllies;
+			List<GameCharacter> npcEnemies;
+			if(allies.contains(npc))
```

Comment:
Allies could be a hashmap with a boolean value, true if ally.

Reply:
Testing for membership in a HashSet should be equally efficient.